mantidproject/mantid
# Don't copy det IDs in Qxy

**Description of work.**
Don't copy detector IDs in QXY, this makes no sense physically and
causes a bug where the first two spectra take the monitor IDs and get
marked as such.

This has been a long standing bug, but was only noticed in Workbench
where monitor spectra are displayed differently (or not at all in colour
fill plots). In Mantid Plot all spectra are plotted identically so this
wasn't noticed


**Report to:** ISIS/Rob (SANS)

**To test:**
- Download the ISIS data training set for LoqDemo
- Open the SANS GUI in workbench
- Set the user file to `MaskFile.txt` in LoqDemo
- Set the batch file to the csv in LoqDemo
- Ensure 2D reduction is selected
- Select the reduced workspace (it's the one not grouped) and show detectors
- The first two spectra should not be marked as monitors
- Colour fill plot, there should not be a large horizontal strip of white at the bottom (this previously took 3/10ths of the screen)

Fixes #29589 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
